### ISSUE TITLE
createTag broken in version >= 4.0.6 for tags longer than 2 characters 

### ISSUE THREAD

**Issue by oserban** [select2/select2#5516]
I tested this code on 4.0.6 and 4.0.6 (only the releases and not the RC) and works well on 4.0.5. My usage of select2 is very simple:

```js
$('#tags').select2({
   tags: true,
   createTag: function (params) {
      console.log("create tag", params)
     return params.term.match(/^[a-z0-9-_]+$/) ? {id: params.term, text: params.term, newTag: true} : null;
     }
});
```

the create tag function is fired for the first 2 characters of the tag and then stops for the rest.

**Comment by jbogdani** [comment]
The issue persists in version 4.0.7.

**Comment by kevin-brown** [comment]
Can you create a jsbin that reproduces this issue?

We don't have this issue in the docs, which makes me suspicious: https://select2.org/tagging

**Comment by oserban** [comment]
Sorry for not providing a JS Bin in the first place. Here it is: https://jsbin.com/yunayiwodo/edit?html,console,output

**Comment by oserban** [comment]
After a bit of digging this seems to be related to #5510 and #5485

**Comment by farzer** [comment]
same problem

**Comment by mariuszkrzaczkowski** [comment]
the same problem and the same lack of solution

**Comment by lucasprogamer** [comment]
same problem

**Comment by lucasprogamer** [comment]
i install version 4.0.0 to solve problem.

**Comment by kuba1999** [comment]
same problem

**Comment by kevin-brown** [comment]
Please see https://github.com/select2/select2/issues/5485#issuecomment-498919585 for why this still remains unsolved.

We have a jsbin for this, which is why we know the issue exists. Unfortunately we can't reproduce this issue in our current test suite, which explains why it wasn't caught earlier. If someone is willing to donate some of their time to making a unit test that reproduces this issue, I'll willing to donate some time to fixing whatever is causing the issue.

**Comment by PHPist** [comment]
This problem is related to jquery-3.4.1.js

**Comment by collimarco** [comment]
@kevin-brown Since this is a severe and widespread issue, can you consider merging the fix now? Several months have already passed. The tests can be added later.

**Comment by Wolfteam** [comment]
This one shoud be reopened.
In this jsfiddle: [Select2](https://jsfiddle.net/wolfteam20/2pwkLq6j/1/)
after you type 2 characters, the first option gets selected

**Comment by kevin-brown** [comment]
This should have been fixed in 4.0.8 with and additional level of edge case smashing in 4.1.0-beta.1 If you are still seeing this issue, please open a new ticket with additional details.
